### PR TITLE
Pornhubpremium

### PR DESCRIPTION
**Pull request by fauresystems**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x ] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x ] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Add pornhubpremium.com to PornHub extractor.

### CODE REVIEW

**Comment by fauresystems** [comment]
PornHub test were broken, so I fixed them and added PornHub Premium login, should be OK now

**Comment by fauresystems** [comment]
Do I need anything to do now?

File: youtube_dl/extractor/extractors.py [hunk 1]
```diff
@@ -900,6 +900,11 @@
     PornHubPlaylistIE,
     PornHubUserVideosIE,
 )
+#from .pornhubpremium import (
+#    PornHubPremiumIE,
+#    PornHubPremiumPlaylistIE,
+#    PornHubPremiumUserVideosIE,
+#)
```
Comment (dstftw) [inline youtube_dl/extractor/extractors.py]: Remove.

File: youtube_dl/extractor/pornhub.py [hunk 2]
```diff
@@ -48,10 +49,12 @@ def dl(*args, **kwargs):
 
 class PornHubIE(PornHubBaseIE):
     IE_DESC = 'PornHub and Thumbzilla'
+    _NETRC_MACHINE = 'pornhubpremium'
+    _LOGIN_URL = 'https://www.pornhubpremium.com/premium/login'
     _VALID_URL = r'''(?x)
                     https?://
                         (?:
-                            (?:[^/]+\.)?(?P<host>pornhub\.(?:com|net))/(?:(?:view_video\.php|video/show)\?viewkey=|embed/)|
+                            (?:[^/]+\.)?(?P<host>(pornhub|pornhubpremium)\.(?:com|net))/(?:(?:view_video\.php|video/show)\?viewkey=|embed/)|
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Don't capture unused groups./

File: youtube_dl/extractor/pornhub.py [hunk 3]
```diff
@@ -160,6 +188,41 @@ def _extract_count(self, pattern, webpage, name):
...
+            self._LOGIN_URL, None, 'Downloading login page')
+
+        login_form = self._hidden_inputs(login_page)
+
+        login_form.update({
+            'username': username,
+            'password': password,
+        })
+
+        self._TOKEN = login_form['token']
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Unused.

File: youtube_dl/extractor/pornhub.py [hunk 4]
```diff
@@ -160,6 +188,41 @@ def _extract_count(self, pattern, webpage, name):
...
+
+        response = self._download_json(
+            "https://www.pornhubpremium.com/front/authenticate", None, 'Logging in',
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Single quotes.

File: youtube_dl/extractor/pornhub.py [hunk 5]
```diff
@@ -74,14 +77,37 @@ class PornHubIE(PornHubBaseIE):
...
+            'like_count': int,
+            'dislike_count': int,
+            'comment_count': int,
+            'age_limit': 18,
+            'tags': list,
+            'categories': list,
+        },
+        'params': {
+            'username': "tdsist",
+            'password': "ZtOUktu0QP",
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Remove credentials from public code and change your pass.

File: .gitignore [hunk 6]
```diff
@@ -51,3 +51,8 @@ venv/
 
 # VS Code related files
 .vscode
+AtomicParsley.exe
+ffmpeg.exe
+ffprobe.exe
+test_fork.py
+venv.bat
```
Comment (dstftw) [inline .gitignore]: Revert all unrelated changes.

**Comment by fauresystems** [comment]
all done :-)

*Everything should be as simple as possible, but not simpler.*
*_______________*
FAURE SYSTEMS
5 square Linné
38320 Eybens

**Comment by fauresystems** [comment]
I'm a github rookie but I guess it's okay now.

**Comment by fauresystems** [comment]
I guess the only test failing is Pornhub Premium because of credentials.

**Comment by dstftw** [comment]
Wrong. See travis log.

**Comment by fauresystems** [comment]
I fall back to previous password, sounds good now.

**Comment by fauresystems** [comment]
Is test going to run automatically again, or I need to do something?

**Comment by fauresystems** [comment]
Test OK on my PC, please tell me what should I do?

Maybe just process Travis again (I changed back the password).

*Everything should be as simple as possible, but not simpler.*
*_______________*
FAURE SYSTEMS
5 square Linné
38320 Eybens

**Comment by fauresystems** [comment]
I don't know what to do with that:
`Traceback (most recent call last):
  File "/home/travis/build/ytdl-org/youtube-dl/test/test_unicode_literals.py", line 52, in test_all_files
    'unicode_literals import  missing in %s' % fn)
  File "/home/travis/build/ytdl-org/youtube-dl/test/helper.py", line 242, in assertRegexpMatches
    self.assertTrue(m, msg)
AssertionError: None is not true : Regexp didn't match: '(?:(?:#.*?|\\s*)\\n)*from __future__ import (?:[a-z_]+,\\s*)*unicode_literals' not found in '"""\ntest_fork.py is not added to git (only for local testing)\n"""\n\nfrom __future__ import unicode_literals\nimport youtube_dl\n\nydl_opts = {\n    \'skip_download\': True,\n}\n\nwith youtube_dl.YoutubeDL(ydl_opts) as ydl:\n    ydl.download([\'http://www.pornhub.com/view_video.php?viewkey=1331683002\'])\n\nexit()\n\nydl_opts = {\n    \'format\': \'bestvideo[ext=mp4]+bestaudio[ext=m4a]\',\n    \'username\': "tdsist",\n    \'password\': "ZtOUktu0QP",\n}\n\nwith youtube_dl.YoutubeDL(ydl_opts) as ydl:\n    ydl.download([\'https://fr.pornhubpremium.com/view_video.php?viewkey=ph5bdbb2cd6df6a\'])', unicode_literals import  missing in /home/travis/build/ytdl-org/youtube-dl/test_fork.py`

**Comment by fauresystems** [comment]
Sounds OK now, you merge the request ?

**Comment by fauresystems** [comment]
Sounds OK now without creds, you merge the request ? Or am I requested to dpo something ?

**Comment by fauresystems** [comment]
Hi Sergey,

I don't see the code in master branch, so not in 2019.07.12 release,

what should I do ?

*Everything should be as simple as possible, but not simpler.*
*_______________*
FAURE SYSTEMS
5 square Linné
38320 Eybens

File: youtube_dl/extractor/pornhub.py [hunk 7]
```diff
@@ -48,10 +49,12 @@ def dl(*args, **kwargs):
 
 class PornHubIE(PornHubBaseIE):
     IE_DESC = 'PornHub and Thumbzilla'
+    _NETRC_MACHINE = 'pornhubpremium'
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: No, should be pornhub. Auth should work for non premium site as well.

File: youtube_dl/extractor/pornhub.py [hunk 8]
```diff
@@ -97,13 +121,14 @@ class PornHubIE(PornHubBaseIE):
         },
     }, {
         # subtitles
-        'url': 'https://www.pornhub.com/view_video.php?viewkey=ph5af5fef7c2aa7',
+        'url': 'https://fr.pornhub.com/view_video.php?viewkey=ph5a73f8c84d1a9',
         'info_dict': {
-            'id': 'ph5af5fef7c2aa7',
+            'id': 'ph5a73f8c84d1a9',
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Remove all unrelated changes.

File: youtube_dl/extractor/pornhub.py [hunk 9]
```diff
@@ -48,10 +49,12 @@ def dl(*args, **kwargs):
 
 class PornHubIE(PornHubBaseIE):
     IE_DESC = 'PornHub and Thumbzilla'
+    _NETRC_MACHINE = 'pornhubpremium'
+    _LOGIN_URL = 'https://www.pornhubpremium.com/premium/login'
     _VALID_URL = r'''(?x)
                     https?://
                         (?:
-                            (?:[^/]+\.)?(?P<host>pornhub\.(?:com|net))/(?:(?:view_video\.php|video/show)\?viewkey=|embed/)|
+                            (?:[^/]+\.)?(?P<host>(?:pornhub|pornhubpremium)\.(?:com|net))/(?:(?:view_video\.php|video/show)\?viewkey=|embed/)|
```
Comment (twaddington) [inline youtube_dl/extractor/pornhub.py]: This would be better as:

```
P<host>pornhub(?:premium)?\.
```
